--- a/docs/conf.py
+++ b/docs/conf.py
@@ -109,11 +109,12 @@
         g.write(subs.xlat(doxy))
 
     try:
-        retcode = subprocess.call(f"cd {folder}; {shutil.which('doxygen')}", shell=True)
+        retcode = subprocess.call("cd {}; {}".format(
+            folder, shutil.which('doxygen')), shell=True)
         if retcode < 0:
-            sys.stderr.write(f"doxygen terminated by signal {-retcode}")
+            sys.stderr.write("doxygen terminated by signal {}".format(-retcode))
     except OSError as e:
-        sys.stderr.write(f"doxygen execution failed: {e}")
+        sys.stderr.write("doxygen execution failed: {}".format(e))
 
 
 def setup(app):
